lib/translate/inquiry.cc

```diff
@@ -0,0 +1,298 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "inquiry.h"
+#include "string.h"
+namespace inquiry {
+// Creates and validates a Inquiry Command struct
+std::optional<scsi_defs::InquiryCommand> raw_cmd_to_scsi_command(
+    absl::Span<const uint32_t> raw_cmd) {
+  if (raw_cmd.empty()) {
+    printf("buffer is empty or nullptr\n");
+    return std::nullopt;
+  }
+
+  uint8_t opcode = raw_cmd[0];
+  if (static_cast<scsi_defs::OpCode>(opcode) != scsi_defs::OpCode::kInquiry) {
+    printf("invalid opcode. expected %ux. got %ux.",
+           static_cast<uint8_t>(scsi_defs::OpCode::kInquiry), opcode);
+    return std::nullopt;
+  }
+
+  // printf("opcode is good, casting\n");
+  // TODO: check invalid parameters in scsi_command
+  return std::make_optional(
+      *reinterpret_cast<const scsi_defs::InquiryCommand*>(&raw_cmd[1]));
+}
+
+// Executes the NVME Identify Controller command
+nvme_defs::IdentifyControllerData nvme_identify_controller() {
```

Comment:
function names should be PascalCase https://google.github.io/styleguide/cppguide.html#Function_Names